src/arrayMethodJoin.js

```diff
@@ -5,7 +5,24 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+
+    if (separator === undefined) {
+      // eslint-disable-next-line no-param-reassign
+      separator = ',';
+    }
```

Comment:
you don't need to mutate argument of function. You can just set default value for the `separator `
```suggestion
```